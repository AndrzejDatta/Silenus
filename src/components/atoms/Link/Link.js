import React from "react";
import PropTypes from "prop-types";
import { StyledLinkElement, StyledIcon } from "./Link.styles";
const Link = ({
  icon,
  size,
  color,
  click,
  path,
  isSearchBar,
  isRight,
  location,
}) => {
  console.log(path);
  return (
    <StyledLinkElement to={path}>
      <StyledIcon
        src={icon}
        size={size}
        color={color}
        onClick={click}
        isSearchBar={isSearchBar}
        isRight={isRight}
        location={location}
      />
    </StyledLinkElement>
  );
};

Link.propTypes = {
  /**size of text inSubtitle atom*/
  size: PropTypes.string.isRequired,
  /**color of Subtitle atom*/
  color: PropTypes.string.isRequired,
  /** icon svg injection*/
  icon: PropTypes.string.isRequired,

  isSearchBar: PropTypes.bool,
};

export default Link;
