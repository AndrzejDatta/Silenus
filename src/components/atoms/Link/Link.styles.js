import { Link } from "react-router-dom";
import styled from "styled-components";
import { theme } from "assets/styles/theme";

export const StyledIcon = styled.img`
  ${(props) => {
    if (props.size === "small") {
      return `
            width: 24px;
            height: 24px; 
        `;
    } else if (props.size === "medium") {
      return `
            width: 27px;
            height: 27px;
            `;
    } else if (props.size === "tiny") {
      return `
            width: 22px;
            height:22px;
            `;
    } else if (props.size === "big") {
      return `
            width:66px;
            height:68px;
            position:absolute;
            `;
    } else if (props.size === "large") {
      return `
            width:86px;
            height:89px;
            position:absolute;
            `;
    } else if (props.size === "great") {
      return `
            width:158px;
            height:164px;
            position:absolute;
            `;
    } else if (props.size === "cameraBig") {
      return `
            width:60px;
            height:35px;
            position:absolute;
            `;
    }
  }};
  ${(props) => {
    if (props.color === "black") {
      return `filter: ${theme.color.filterBlack}`;
    } else if (props.color === "white") {
      return `filter: ${theme.color.filterWhite}`;
    } else if (props.color === "green") {
      return `filter: ${theme.color.filterGreen}`;
    } else if (props.color === "lightGreen") {
      return `filter: ${theme.color.filterLightGreen}`;
    }
  }};
  ${(props) => {
    switch (props.location) {
      case 1:
        return `
        top:20%;
        left:-17%;
        `;

      case 2:
        return `
        top:47%;
        left:0%;
        transform: scaleX(-1);
        `;
      case 3:
        return `
        top:15%;
        right:3%;
        
        `;
      case 4:
        return `
        top:43%;
        right:-20%;
        transform: scaleX(-1);
        `;
      default:
    }
    if (props.location === 1) {
      return `
      top:20%;
      left:-5%;
      
      `;
    } else if (props.color === "white") {
      return `filter: ${theme.color.filterWhite}`;
    } else if (props.color === "green") {
      return `filter: ${theme.color.filterGreen}`;
    }
  }};

  ${(props) => {
    if (props.isSearchBar && props.isRight) {
      return `
      position:absolute;
      right:10px;
      top:50%;
      transform:translateY(-50%);
      `;
    } else if (props.isSearchBar) {
      return `
     position:absolute;
     left:10px;
     top:50%;
     transform:translateY(-50%);
      
      `;
    }
  }};
  cursor: pointer;
`;

export const StyledLinkElement = styled(Link)`
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  justify-self: flex-end;
  padding: 5px 0px;
`;
