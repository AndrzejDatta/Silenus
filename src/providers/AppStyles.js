import { ThemeProvider } from 'styled-components';
import { theme } from 'assets/styles/theme';
import { GlobalStyle } from 'assets/styles/GlobalStyles';
export const AppStyles = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};
