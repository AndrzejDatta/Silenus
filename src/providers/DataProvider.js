import { createContext, useState } from "react";
import { dataPL } from "assets/data/data";

export const dataContext = createContext({
  startPage: {},
  loginPage: {},
  Dashboard: {},
  MojeDoniczki: {},
  DodajDoniczke: {},
  MojeRośliny: {},
  KalendarzOpieki: {},
  KontoUżytkownika: {},
  ZidentyfikujRośline: {},
  PotDetails: {},
  PlantDetails: {},
  Navbar: {},
  images: {},
  navIcons: {},
  potIcons: {},
  categoryIcons: {},
  navIconsColor: [],
  setNavIconColor: () => {},
  changeColor: () => {},
  importAll: () => {},
});

const objectName = [
  "startPage",
  "loginPage",
  "Dashboard",
  "MojeDoniczki",
  "DodajDoniczke",
  "MojeRośliny",
  "KalendarzOpieki",
  "KontoUżytkownika",
  "ZidentyfikujRośline",
  "PotDetails",
  "PlantDetails",
  "Navbar",
];

export const DataProvider = ({ children }) => {
  for (let i = 0; i < dataPL.length; i++) {
    objectName[i] = { ...dataPL[i] };
  }
  const [
    startPage,
    loginPage,
    Dashboard,
    MojeDoniczki,
    DodajDoniczke,
    MojeRośliny,
    KalendarzOpieki,
    KontoUżytkownika,
    ZidentyfikujRośline,
    PotDetails,
    PlantDetails,
    Navbar,
  ] = objectName;

  const [navIconsColor, setNavIconColor] = useState([
    {
      name: "plant",
      src: "potted_plant.svg",
      isClicked: false,
    },
    {
      name: "calendar",
      src: "calendar.svg",
      isClicked: false,
    },
    {
      name: "camera",
      src: "camera.svg",
      isClicked: false,
    },
    {
      name: "user",
      src: "accountCircle.svg",
      isClicked: false,
    },
  ]);

  function importAll(r) {
    let images = {};
    r.keys().map((item, index) => {
      images[item.replace("./", "")] = r(item);
    });
    return images;
  }

  const images = importAll(
    require.context(
      "assets/icons/userAccountOption",
      false,
      /\.(png|jpe?g|svg)$/
    )
  );
  const navIcons = importAll(
    require.context("assets/icons/navIcons", false, /\.(png|jpe?g|svg)$/)
  );

  const potIcons = importAll(
    require.context("assets/icons/potStatus", false, /\.(png|jpe?g|svg)$/)
  );

  const categoryIcons = importAll(
    require.context("assets/icons/categoryIcons", false, /\.(png|jpe?g|svg)$/)
  );

  function changeColor(nameElement) {
    setNavIconColor((prevData) =>
      prevData.map((item) => {
        if (item.name === nameElement && !item.isClicked) {
          return {
            ...item,
            isClicked: true,
          };
        } else {
          return {
            ...item,
            isClicked: false,
          };
        }
      })
    );
  }

  return (
    <dataContext.Provider
      value={{
        startPage,
        loginPage,
        Dashboard,
        DodajDoniczke,
        MojeDoniczki,
        MojeRośliny,
        KalendarzOpieki,
        KontoUżytkownika,
        ZidentyfikujRośline,
        PotDetails,
        PlantDetails,
        Navbar,
        images,
        navIcons,
        potIcons,
        categoryIcons,
        navIconsColor,
        setNavIconColor,
        changeColor,
        importAll,
      }}
    >
      {children}
    </dataContext.Provider>
  );
};

//TODO:improve way
