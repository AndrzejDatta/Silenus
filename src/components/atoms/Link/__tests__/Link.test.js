import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { AppStyles } from "providers/AppStyles";
import { render, screen } from "@testing-library/react";
import Link from "components/atoms/Link/Link";
import { MemoryRouter } from "react-router-dom";

describe("Test Link Component", () => {
  it("Test default style", () => {
    const tree = renderer
      .create(
        <AppStyles>
          <MemoryRouter>
            <Link />
          </MemoryRouter>
        </AppStyles>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
