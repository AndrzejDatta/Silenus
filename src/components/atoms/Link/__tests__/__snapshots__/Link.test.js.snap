// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Link Component Test default style 1`] = `
.c1 {
  width: 24px;
  height: 24px;
  -webkit-filter: invert(57%) sepia(93%) saturate(279%) hue-rotate(113deg) brightness(82%) contrast(87%);
  filter: invert(57%) sepia(93%) saturate(279%) hue-rotate(113deg) brightness(82%) contrast(87%);
  -webkit-filter: invert(57%) sepia(93%) saturate(279%) hue-rotate(113deg) brightness(82%) contrast(87%);
  filter: invert(57%) sepia(93%) saturate(279%) hue-rotate(113deg) brightness(82%) contrast(87%);
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-self: flex-end;
  padding: 5px 0px;
}

<a
  className="c0"
  href="/dashboard"
  onClick={[Function]}
>
  <img
    className="c1"
    color="green"
    onClick={[Function]}
    size="small"
    src=""
  />
</a>
`;
